### PR TITLE
OCLOMRS-331: Add custom error messages when fetching the API while offline

### PR DESCRIPTION
**Pull request by Duncanian**
# JIRA TICKET NAME:
[Add custom error messages when fetching the API while offline](https://issues.openmrs.org/browse/OCLOMRS-331)

# Summary:
Currently, there are no error messages when trying to access the API when offline hence providing a descriptive information makes the user aware that they are offline at that time.

### CODE REVIEW

File: src/redux/actions/concepts/dictionaryConcepts.js [hunk 1]
```diff
@@ -142,10 +142,11 @@ export const fetchDictionaryConcepts = (
       dispatch(populateSidenav());
     }
   } catch (error) {
-    if (error.response) {
+    if (error.response !== undefined) {
       dispatch(isErrored(error.response.data, FETCH_DICTIONARY_CONCEPT));
       dispatch(isFetching(false));
     }
+    notify.show('Network Error. Please try again later!', 'error', 3000);
```
Comment (ozezzy) [inline src/redux/actions/concepts/dictionaryConcepts.js]: I think you can use the function `showNetworkError();` here.

Comment (Duncanian) [reply src/redux/actions/concepts/dictionaryConcepts.js]: working on it

Comment (Duncanian) [reply src/redux/actions/concepts/dictionaryConcepts.js]: @ozezzy , this file doesn't contain the `showNetworkError` function used anywhere.

File: src/redux/actions/auth/authActions.js [hunk 2]
```diff
@@ -19,10 +19,10 @@ const loginAction = ({ username, password }) => (dispatch) => {
     localStorage.setItem('username', `${username}`);
     return dispatch(login(response));
   }).catch((error) => {
-    if (error.response) {
-      return dispatch(loginFailed(error.response.data.detail));
+    if (error.response === undefined) {
+      return dispatch(loginFailed('Network Error. Please try again later!'));
```
Comment (dkayiwa) [inline src/redux/actions/auth/authActions.js]: Can you use a constant for this error message?

Comment (Duncanian) [reply src/redux/actions/auth/authActions.js]: done

File: src/redux/actions/auth/authActions.js [hunk 3]
```diff
@@ -19,10 +19,11 @@ const loginAction = ({ username, password }) => (dispatch) => {
     localStorage.setItem('username', `${username}`);
     return dispatch(login(response));
   }).catch((error) => {
-    if (error.response) {
-      return dispatch(loginFailed(error.response.data.detail));
+    const offlineError = 'Network Error. Please try again later!';
```
Comment (dkayiwa) [inline src/redux/actions/auth/authActions.js]: Don't we already have this here? https://github.com/openmrs/openmrs-ocl-client/search?q=showNetworkError&unscoped_q=showNetworkError

Comment (dkayiwa) [reply src/redux/actions/auth/authActions.js]: Does that mean we are free to duplicate the same thing across multiple files?

Comment (Duncanian) [reply src/redux/actions/auth/authActions.js]: Maybe am getting it wrong, so should I import it from the other file?

Comment (dkayiwa) [reply src/redux/actions/auth/authActions.js]: I do not want to dictate to you in regards to how you should address it. Am simply raising the important software engineering principle of not duplicating code. Are you aware of it?

Comment (Duncanian) [reply src/redux/actions/auth/authActions.js]: Yeah, am aware of it. So I bet importing and reusing it serves that principle well.

Comment (Duncanian) [reply src/redux/actions/auth/authActions.js]: I have reused it.

File: src/redux/actions/concepts/dictionaryConcepts.js [hunk 4]
```diff
@@ -142,10 +143,11 @@ export const fetchDictionaryConcepts = (
       dispatch(populateSidenav());
     }
   } catch (error) {
-    if (error.response) {
+    if (error.response !== undefined) {
       dispatch(isErrored(error.response.data, FETCH_DICTIONARY_CONCEPT));
       dispatch(isFetching(false));
     }
+    showNetworkError();
```
[the remark this answers is not in the document]
Comment (Duncanian) [reply src/redux/actions/concepts/dictionaryConcepts.js]: It's in the else part of the statement

[the remark this answers is not in the document]
Comment (dkayiwa) [reply src/redux/actions/concepts/dictionaryConcepts.js]: How come that here you put it in the else, where elsewhere you put it under? if error.response === undefined

[the remark this answers is not in the document]
Comment (Duncanian) [reply src/redux/actions/concepts/dictionaryConcepts.js]: When I put it in the else section, it works better with no errors. Furthermore, the implementation remains the same and tests pass really well.

[the remark this answers is not in the document]
Comment (dkayiwa) [reply src/redux/actions/concepts/dictionaryConcepts.js]: I do not think you are getting my question. Can you read it again?

[the remark this answers is not in the document]
Comment (Duncanian) [reply src/redux/actions/concepts/dictionaryConcepts.js]: Oh, I got it, It is the same implementation like this one, so I should avoid using the else and instead put it under.

**Comment by collinewait** [comment]
Resolve conflicts

**Comment by Duncanian** [comment]
@collinewait , I have resolved them.

**Pull request merged**